src/waypoint.jsx

```diff
@@ -328,7 +332,7 @@ export default class Waypoint extends React.Component {
   render() {
     // We need an element that we can locate in the DOM to determine where it is
     // rendered relative to the top of its context.
-    return <span style={{fontSize: 0}} />;
+    return <span ref={this.refElement} style={{fontSize: 0}} />;
```

Comment:
This is fine, but I'm not sure it really makes much difference either way. My preference would be to use the arrow function here to reduce unnecessary indirection.